TMS_API/Controllers/EventController.cs

```diff
@@ -0,0 +1,110 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Microsoft.IdentityModel.Tokens;
+using System.ComponentModel;
+using System.Net.WebSockets;
+using System.Runtime.Serialization;
+using TMS_API.Exceptions;
+using TMS_API.Models;
+using TMS_API.Models.DTO;
+using TMS_API.Repositories;
+
+namespace TMS_API.Controllers
+{
+    [Route("api/[controller]/[action]")]
+    [ApiController]
+    public class EventController : ControllerBase
+    {
+        private readonly IEventRepository _eventRepository;
+        private readonly IMapper _mapper;
+
+        public EventController(IEventRepository eventRepository, IMapper mapper) 
+        {
+            _eventRepository = eventRepository;
+            _mapper = mapper;
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<List<EventDTO>>> GetAll()
+        {
+            var events = await _eventRepository.GetAll();
```

Comment:
It is recommended not to call repository methods directly in controller, It is better to have a TMS.BusinessLogic  project, that contains references to repositories. See 3 layer architecture: https://medium.com/@deanrubin/the-three-layered-architecture-fe30cb0e4a6